pkg/cli/buildserver.go

```diff
@@ -47,15 +48,27 @@ func (s *BuildServer) Run(cmd *cobra.Command, args []string) error {
 	server := buildserver.NewServer(s.UUID, s.Namespace, pubKey, privKey, c)
 	address := fmt.Sprintf("0.0.0.0:%d", s.ListenPort)
 
+	var p *tcpproxy.Proxy
 	if s.ForwardService != "" {
-		var p tcpproxy.Proxy
+		p = new(tcpproxy.Proxy)
 		p.AddRoute(fmt.Sprintf(":%d", s.ForwardPort), tcpproxy.To(s.ForwardService))
 		go func() {
 			logrus.Infof("Forwarding :%d to %s", s.ForwardPort, s.ForwardService)
 			logrus.Fatal(p.Run())
 		}()
 	}
 
+	go func() {
+		// Nothing here is using the context. Therefore, when the context is canceled, we should just exit.
+		<-cmd.Context().Done()
+		if p != nil {
+			if err = p.Close(); err != nil {
+				logrus.Errorf("Error closing proxy: %v", err)
+			}
+		}
+		os.Exit(0)
+	}()
```

Comment:
I don't think we need need to use a pointer for `p` since both closures capture it in scope:

```go
	var p tcpproxy.Proxy
	if s.ForwardService != "" {
		p.AddRoute(fmt.Sprintf(":%d", s.ForwardPort), tcpproxy.To(s.ForwardService))
		go func() {
			logrus.Infof("Forwarding :%d to %s", s.ForwardPort, s.ForwardService)
			logrus.Fatal(p.Run())
		}()
	}

	go func() {
		// Nothing here is using the context. Therefore, when the context is canceled, we should just exit.
		<-cmd.Context().Done()
	    if err = p.Close(); err != nil {
			logrus.Errorf("Error closing proxy: %v", err)
		}

		os.Exit(0)
	}()
```